src/dune_engine/action.ml

```diff
@@ -313,7 +316,11 @@ let is_useful_to distribute memoize =
     | Dynamic_run _ -> true
     | System _ -> true
     | Bash _ -> true
-    | Format_dune_file _ -> memoize
+    | Extension (_, { how_to_cache; _ }) -> (
+      match how_to_cache with
+      | Memoize_or_distribute.Neither -> false
```

Comment:
I don't fully understand what's going on here, but it doesn't seem right.
1) this implementation makes `is_useful_to_sandbox` always return `false` on extension actions.
2) The assumption in `is_useful_to` seems to be that it can't be useful to sandbox an action if it's not useful to "memoize" it. (the term memoize is confusing to me, but let's put that aside for now) This assumption seems very strange to me, in fact in jenga it's an allowable configuration to have a sandboxed action that's not stored in the shared cache.

Not sure if we should do anything about (2) at this point, but (1) seems clearly not right (e.g. cram is definitely sandboxable).

Reply:
One idea might be to just use (another) closure that mimicks the structure of `is_useful_to` in general, so `how_to_cache` gets type `'a -> bool -> bool -> bool`.

Reply:
Please let's not perpetuate this inscrutable type.
To be honest, I think it's fine to just rip this out and return `true` in all cases. After all, "false" comes with a burden of proof, but "true" is supposed to be a safe answer here.

Reply:
If we're keeping the customizability and going for some cleanup, I could go with something like this:

```
type is_useful_to_cache = Workspace_only | Host_only | Distributed
...
  is_useful_to_sandbox : is_useful
  is_useful_to_cache : is_useful_to_cache
...
```

but again, I think it's not worth it and simply returning `true` is better.

Reply:
One concern I had was that the actions I was hoping to change to extensions had varying responses to this function, and I'd like to preserve the existing behavior to some extent.

For example, `Format_dune_file` returns `memoize` but `Cram` returns `true`. I agree that this `is_useful_to` type is difficult to understand and that we might want to do something better.

Reply:
Oh, I see. 
I don't know how important it is that dune file formatting is not sandboxed, but fair enough.
I guess the minimum change is to add a fourth constructor `Memoize_or_distribute.Distribute_and_maybe_sandbox` or something ridiculous like that, but I would rather just split the `is_useful_to_sandbox` from `is_useful_to_cache` completely (just duplicate the pattern-match) and then things should become more sensible.

Reply:
Would this necessitate another field (`can_sandbox`?) in the extension spec?

Reply:
Yes, but it's not `can_sandbox`, it's `useful_to_sandbox`. It's assumed that the actions can be sandboxed regardless of this value (e.g. if they are combined with another action that's useful to sandbox). If the action can't be sandboxed, there is another mechanism to specify that (via its deps).

Reply:
Been changed.